stats/influxdb/collector.go

```diff
@@ -107,11 +107,31 @@ func (c *Collector) commit() {
 		return
 	}
 
+	cache := map[*stats.SampleTags]struct {
+		tags   map[string]string
+		values map[string]interface{}
+	}{}
+
+	var tags map[string]string
+	var values map[string]interface{}
 	for _, sample := range samples {
+		if cached, ok := cache[sample.Tags]; ok {
+			tags = cached.tags
+			values = cached.values
+			values["value"] = sample.Value // Overwrite the "value" field
```

Comment:
I think that this would be a bug that causes all `values["value"]` to be the same: https://play.golang.org/p/xikPsAxZWFI
Basically assigning a map with `=` doesn't do a deep copy of the values and changing the "copy" affects the original :/ One of those unpleasant Go gotchas... 
You probably have to create a new map every time in the loop (`values := map[string]interface{}{"value": sample.Value}`) and then copy all items from `cached.values` in it with a loop.

Reply:
Hmm actually it's probably a better idea to create a new *empty* map, copy all `cached.values` in it and then set `values["value"]` to `sample.Value`, just in case someone tries to create a new tag named `"value"` :smile:

Reply:
This actually works but as you said in private chat, I will change it to be safe if InfluxDB changes it's implementation and start caching/batching stuff internally.